books/books/spiders/followall.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import re
+from six.moves.urllib.parse import urlparse
+import datetime
+import scrapy
+from scrapy.http import Request, HtmlResponse
+from scrapy.linkextractors import LinkExtractor
+import click
+from books.items import Page
+t = datetime.datetime.utcnow()
+
+
+class FollowAllSpider(scrapy.Spider):
+
+    name = 'followall'
+    ratings_map = {
+        'one': 1,
+        'two': 2,
+        'three': 3,
+        'four': 4,
+        'five': 5,
+    }
+
+    def __init__(self, **kw):
+        super(FollowAllSpider, self).__init__(**kw)
+        url = 'http://localhost/books.toscrape.com/index.html'
+        if not url.startswith('http://') and not url.startswith('https://'):
+            url = 'http://%s/' % url
+        self.url = url
+        self.allowed_domains = [re.sub(r'^www\.', '', urlparse(url).hostname)]
+        self.link_extractor = LinkExtractor()
+        self.cookies_seen = set()
+        self.previtem = 0
+        self.items = 0
+        self.timesec = datetime.datetime.utcnow()
+
+    def start_requests(self):
+        return [Request(self.url, callback=self.parse, dont_filter=True)]
+
+    def parse(self, response):
+        """Parse a PageItem and all requests to follow"""
+        page = self._get_item(response)
+        r = [page]
+
+        r.extend(self._extract_requests(response))
+        self.items = self.crawler.stats.get_value('item_scraped_count', 0)
+        pages = self.crawler.stats.get_value('response_received_count', 0)
+        a = self.crawler.stats.get_value('start_time')
+        b = datetime.datetime.utcnow()
+        self.timesec = b - a
+
+        if 280 <= self.items <= 300:
```

Comment:
Ah, right, now I understand - each page contains multiple items, so it's not exactly 300.

If it does improve stability, maybe using a number of pages instead of number of items would be cleaner, cause it won't need an inequality that might break if more items are extracted.